xml/System.Reflection/AssemblyVersionAttribute.xml

```diff
@@ -86,12 +86,32 @@
   
 > [!IMPORTANT]
 >  All components of the version must be integers greater than or equal to 0. Metadata restricts the major, minor, build, and revision components for an assembly to a maximum value of <xref:System.UInt16.MaxValue?displayProperty=nameWithType> - 1. If a component exceeds this value, a compilation error occurs.  
-  
- You can specify all the values or you can accept the default build number, revision number, or both by using an asterisk (\*). For example, `[assembly:AssemblyVersion("2.3.25.1")]` indicates 2 as the major version, 3 as the minor version, 25 as the build number, and 1 as the revision number. A version number such as `[assembly:AssemblyVersion("1.2.*")]` specifies 1 as the major version, 2 as the minor version, and accepts the default build and revision numbers. A version number such as `[assembly:AssemblyVersion("1.2.15.*")]` specifies 1 as the major version, 2 as the minor version, 15 as the build number, and accepts the default revision number. The default build number increments daily. The default revision number is the number of seconds since midnight local time (without taking into account time zone adjustments for daylight saving time), divided by 2.  
-  
-> [!NOTE]
->  If you specify an asterisk for the build number, you cannot specify a revision number.  
-  
+
+ For example, `[assembly:AssemblyVersion("2.3.25.1")]` indicates 2 as the major version, 3 as the minor version, 25 as the build number, and 1 as the revision number. 
+
+ The <xref:System.Reflection.AssemblyVersionAttribute> attribute allows to specify an asterisk (\*) in place of the build or revision number. A version number such as `[assembly:AssemblyVersion("1.2.*")]` specifies 1 as the major version, 2 as the minor version, and accepts the default build and revision numbers. A version number such as `[assembly:AssemblyVersion("1.2.15.*")]` specifies 1 as the major version, 2 as the minor version, 15 as the build number, and accepts the default revision number. The default build number increments daily. The default revision number is the number of seconds since midnight local time (without taking into account time zone adjustments for daylight saving time), divided by 2. If you specify an asterisk for the build number, you cannot specify a revision number.  
+
+> [!IMPORTANT] 
+> Use of <xref:System.Reflection.AssemblyVersionAttribute> attribute that specifies an asterisk makes the build outputs non-reproducible (see [Reproducible Builds](https://reproducible-builds.org/)).
```

Comment:
```suggestion
> Use of the <xref:System.Reflection.AssemblyVersionAttribute> attribute that specifies an asterisk:
>
> - Makes the build outputs non-reproducible (see [Reproducible builds](https://reproducible-builds.org/)).
```

Reply:
This is correct but the situation is more guarded than the docs suggest. Builds are only reproducible if you have `<Deterministic>true</Deterministic>` . That is the default in the .NET SDK  so all modern builds have this set. If that is set and you use `*` in versions then you will get a diagonstic. For example: 

```cmd
E:\temp\console\.complog\console\export\src\obj\Debug\net9.0\console.AssemblyInfo.cs(19,55): error CS8357: The specified version string '1.0.0.*' contains wildcards, which are not compatible with determinism. Either remove wildcards from the version string, or disable determinism for this compilation
```

Reply:
Does build caching in VS require Deterministic=true?

Reply:
Seems like `CS8357` is not documented: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/CS8357